src/AWS.Deploy.Orchestration/Utilities/DeployedApplicationQueryer.cs

```diff
@@ -204,10 +225,37 @@ private async Task<List<CloudApplication>> GetExistingBeanstalkEnvironments()
                 if (tags.Any(x => string.Equals(x.Key, Constants.CloudFormationIdentifier.STACK_TAG)))
                     continue;
 
-                validEnvironments.Add(new CloudApplication(env.EnvironmentName, env.EnvironmentArn, CloudApplicationResourceType.BeanstalkEnvironment, Constants.RecipeIdentifier.EXISTING_BEANSTALK_ENVIRONMENT_RECIPE_ID, env.DateUpdated));
+                validEnvironments.Add(new CloudApplication(env.EnvironmentName, env.EnvironmentId, CloudApplicationResourceType.BeanstalkEnvironment, Constants.RecipeIdentifier.EXISTING_BEANSTALK_ENVIRONMENT_RECIPE_ID, env.DateUpdated));
             }
 
             return validEnvironments;
         }
+
+        private async Task<IDictionary<string, object>> GetBeanstalkEnvironmentConfigurationSettings(string environmentName)
+        {
+            IDictionary<string, object> optionSettings = new Dictionary<string, object>();
+            var configurationSettings = await _awsResourceQueryer.GetBeanstalkEnvironmentConfigurationSettings(environmentName);
+
+            foreach (var (optionId, optionNameSpace, optionName) in Constants.ElasticBeanstalk.OptionSettingQueryList)
+            {
+                var configurationSetting = GetBeanstalkEnvironmentConfigurationSetting(configurationSettings, optionNameSpace, optionName);
+
+                if (string.IsNullOrEmpty(configurationSetting?.Value))
+                    continue;
+
+                optionSettings[optionId] = configurationSetting.Value;
+            }
+
+            return optionSettings;
+        }
+
+        private ConfigurationOptionSetting GetBeanstalkEnvironmentConfigurationSetting(List<ConfigurationOptionSetting> configurationSettings, string optionNameSpace, string optionName)
+        {
+            var configurationSetting = configurationSettings
+                .Where(x => string.Equals(optionNameSpace, x.Namespace) && string.Equals(optionName, x.OptionName))
```

Comment:
nit: you don't need to use Where. You can put the condition inside FirstOrDefault